iothub_client/samples/pnp/pnp_temperature_controller/pnp_temperature_controller.c

```diff
@@ -5,57 +5,48 @@
-// IoTHub Device Client and IoT core utility related header files
+// JSON parser library.
+#include "parson.h"
+
+// IoT Hub device client and IoT core utility related header files.
 #include "iothub.h"
 #include "iothub_device_client_ll.h"
 #include "iothub_client_options.h"
+#include "iothubtransportmqtt.h"
 #include "iothub_message.h"
-#include "azure_c_shared_utility/strings.h"
+#include "iothub_client_properties.h"
 #include "azure_c_shared_utility/threadapi.h"
 #include "azure_c_shared_utility/xlogging.h"
+#include "iothubtransportmqtt.h"
 
-// PnP utilities.
-#include "pnp_device_client_ll.h"
-#include "pnp_protocol.h"
-
-// Headers that provide implementation for subcomponents (the two thermostat components and DeviceInfo)
-#include "pnp_thermostat_component.h"
-#include "pnp_deviceinfo_component.h"
+#include "pnp_status_values.h"
 
-// Environment variable used to specify how app connects to hub and the two possible values
-static const char g_securityTypeEnvironmentVariable[] = "IOTHUB_DEVICE_SECURITY_TYPE";
-static const char g_securityTypeConnectionStringValue[] = "connectionString";
-static const char g_securityTypeDpsValue[] = "DPS";
+#ifdef SET_TRUSTED_CERT_IN_SAMPLES
+// For devices that do not have (or want) an OS level trusted certificate store,
+// but instead bring in default trusted certificates from the Azure IoT C SDK.
+#include "azure_c_shared_utility/shared_util_options.h"
+#include "certs.h"
+#endif // SET_TRUSTED_CERT_IN_SAMPLES
 
-// Environment variable used to specify this application's connection string
-static const char g_connectionStringEnvironmentVariable[] = "IOTHUB_DEVICE_CONNECTION_STRING";
-
-// Values of connection / security settings read from environment variables and/or DPS runtime
-PNP_DEVICE_CONFIGURATION g_pnpDeviceConfiguration;
+#include "pnp_sample_config.h"
 
 #ifdef USE_PROV_MODULE_FULL
-// Environment variable used to specify this application's DPS id scope
-static const char g_dpsIdScopeEnvironmentVariable[] = "IOTHUB_DEVICE_DPS_ID_SCOPE";
-
-// Environment variable used to specify this application's DPS device id
-static const char g_dpsDeviceIdEnvironmentVariable[] = "IOTHUB_DEVICE_DPS_DEVICE_ID";
-
-// Environment variable used to specify this application's DPS device key
-static const char g_dpsDeviceKeyEnvironmentVariable[] = "IOTHUB_DEVICE_DPS_DEVICE_KEY";
+#include "pnp_dps_ll.h"
+#endif // USE_PROV_MODULE_FULL
 
-// Environment variable used to optionally specify this application's DPS id scope
-static const char g_dpsEndpointEnvironmentVariable[] = "IOTHUB_DEVICE_DPS_ENDPOINT";
+// Headers that provide implementation for subcomponents (the two thermostat components and DeviceInfo)
+#include "pnp_thermostat_component.h"
+#include "pnp_deviceinfo_component.h"
 
-// Global provisioning endpoint for DPS if one is not specified via the environment
-static const char g_dps_DefaultGlobalProvUri[] = "global.azure-devices-provisioning.net";
-#endif
+#define CURRENT_WORKING_SET_BUFFER_SIZE 64
```

Comment:
>CURRENT_WORKING_SET_BUFFER_SIZE [](http://example.com/codeflow?start=8&length=31)

how about making these const int or converting all global const like on line 56 to macro?

Reply:
Yeah I generally prefer const over preprocessor too.  But here -- 

The difference is `g_sleepBetweenPollsMs` on 56 is just used as a parameter in a function call.  But the `CURRENT_WORKING_SET_BUFFER_SIZE` is used when initializing the max size of a static array.

I haven't researched, but I'm almost positive I've seen at least some C99 compilers get very unhappy if an array size is set with a variable - even a const - so it demands the pre-processor come in and stitch it up 1st.